### PR TITLE
[CHI-162] Update step component with horizontal labels

### PR DESCRIPTION
**Pull request by davidojedacl**
Pending resolve problem with IE11 with links position

### CODE REVIEW

File: src/chi/components/steps/steps.scss [hunk 1]
```diff
@@ -87,5 +97,100 @@
...
+            }
+          }
+        }
+
+        & a {
+          color: $brand-color;
+        }
+      }
+
+      &.-horizontalLabels {
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: @davidojedacl @jllr Horizontal labels shouldn't be dependent on the -inverse class. It's a alternate layout so we should support both white and black backgrounds.

File: src/chi/components/steps/steps.scss [hunk 2]
```diff
@@ -1,8 +1,18 @@
 @import 'variables';
 @import 'mixins';
 
-.chi {
+@font-face {
+  font-family: 'chi';
+  src: url('webfont/chi.eot');
+}
+@font-face {
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: @davidojedacl @jllr Why is this needed?

File: src/chi/components/steps/steps.scss [hunk 3]
```diff
@@ -87,5 +97,100 @@
...
+
+        & > li {
+          &::before {
+            background-color: set-color(black);
+            border: 0.125rem solid $brand-color;
+            counter-increment: step;
+            content: counter(step);
+            height: 1.3rem;
+            margin: 0 auto;
+            width: 1.3rem;
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: Add `font-weight: 600;` and change font-size of numbers to `font-size: 0.8125rem;`

File: src/chi/components/steps/steps.scss [hunk 4]
```diff
@@ -87,5 +97,100 @@
...
+            }
+
+            & ~ li {
+              &::before {
+                background-color: set-color(black);
+                border-color: set-color(grey, 40);
+              }
+
+              & a {
+                color: $muted-color;
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: Change value to `set-color(grey, 40);`

File: src/chi/components/steps/steps.scss [hunk 5]
```diff
@@ -87,5 +97,100 @@
...
+
+        &.-active {
+          & ~ li {
+            &::before {
+              background-color: set-color(grey, 40);
+              border-color: set-color(black);
+            }
+
+            & a {
+              color: $muted-color;
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: Change value to `set-color(grey, 40);`

File: src/chi/components/steps/steps.scss (hunk 2 above)
Comment (davidojedacl) [reply src/chi/components/steps/steps.scss]: You'are right. Not is necessary.

File: src/chi/components/steps/steps.scss [hunk 6]
```diff
@@ -87,5 +92,131 @@
...
+        padding-left: 0.5rem;
+        padding-right: 0.5rem;
+        position: absolute;
+        top: 0.125rem;
+      }
+    }
+
+    &.-inverse {
+      background-color: set-color(black);
+      padding: 1rem;
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl this padding can be removed as it is duplicated from .-horizontalLabels

File: src/chi/components/steps/steps.scss [hunk 7]
```diff
@@ -87,5 +92,131 @@
         color: $brand-color;
       }
     }
+
+    &.-horizontalLabels {
+      background-color: set-color(white);
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl remove this background-color as it must be transparent in the base scenario

File: src/chi/components/steps/steps.scss [hunk 8]
```diff
@@ -87,5 +92,131 @@
...
+          }
+
+          & a {
+            margin-left: 0.75rem;
+          }
+        }
+
+        &.-completed {
+          &::before {
+            border: .125rem solid set-color($brand-color);
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl remove this property as it is duplicated from .-horizontalLabels li:before

File: src/chi/components/steps/steps.scss [hunk 9]
```diff
@@ -87,5 +92,131 @@
...
+
+        & a {
+          color: $brand-color;
+        }
+      }
+
+      &.-horizontalLabels {
+        & > li {
+          &::before {
+            background-color: set-color(black);
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl please add border:.125rem solid #00853f

File: src/chi/components/steps/steps.scss [hunk 10]
```diff
@@ -87,5 +92,131 @@
...
+          }
+
+          &.-active {
+            &::before {
+              border: .125rem solid set-color($brand-color);
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl please remove this property as it is duplicated

File: src/chi/components/steps/steps.scss [hunk 11]
```diff
@@ -87,5 +91,127 @@
...
+      & > li {
+        &::before {
+          background-color: $brand-color;
+          border: 0.125rem solid set-color(black);
+        }
+
+        &.-active {
+          & ~ li {
+            &::before {
+              background-color: set-color(grey, 40);
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl remove this property as it is duplicated

File: src/chi/components/steps/steps.scss [hunk 12]
```diff
@@ -87,5 +91,127 @@
...
+        top: 0.125rem;
+      }
+    }
+
+    &.-inverse {
+      background-color: set-color(black);
+
+      & > li {
+        &::before {
+          background-color: $brand-color;
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl remove this property as it is duplicated

File: src/chi/components/steps/steps.scss [hunk 13]
```diff
@@ -87,5 +91,127 @@
...
+        & a {
+          color: $brand-color;
+        }
+      }
+
+      &.-horizontalLabels {
+        & > li {
+          &::before {
+            background-color: set-color(black);
+            border: 0 0 0 0.125rem solid $brand-color;
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl remove the first three zero's as they are not necessary

File: src/chi/components/steps/steps.scss [hunk 14]
```diff
@@ -87,5 +91,125 @@
...
+      background-color: set-color(black);
+
+      & > li {
+        &::before {
+          border: 0.125rem solid set-color(black);
+        }
+
+        &.-active {
+          & ~ li {
+            &::before {
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl this :before is not necessary, please remove it

File: src/chi/components/steps/steps.scss [hunk 15]
```diff
@@ -87,5 +91,125 @@
...
+
+          &.-active {
+            &::before {
+              box-shadow: 0 0 0 0.125rem set-color(black);
+            }
+
+            & ~ li {
+              &::before {
+                background-color: set-color(black);
+                border-color: set-color(grey, 40);
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl please remove this property

File: src/chi/components/steps/steps.scss [hunk 16]
```diff
@@ -87,5 +87,120 @@
...
+
+      & > li {
+        &::before {
+          border: 0.125rem solid set-color(black);
+        }
+
+        &.-active {
+          & ~ li {
+            & a {
+              color: set-color(grey, 40);
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl please remove this line

File: src/chi/components/steps/steps.scss [hunk 17]
```diff
@@ -87,5 +87,120 @@
...
+          }
+
+          & ~ li {
+            &::before {
+              background-color: set-color(white);
+              border-color: set-color(grey, 40);
+            }
+
+            & a {
+              color: set-color(grey, 40);
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl please remove this line

**Comment by davidojedacl** [comment]
@mattnickles @jllr  I'm studying other possibilities without a fixed background but with a grid i think will be very difficult. Have you seen something similar with grid?
